### PR TITLE
 (odsp-client): Interface cleanup and pass itemId to getContainer

### PR DESCRIPTION
**Pull request by sonalideshpandemsft**
Couple of feedback which are included in this PR are as follows:

1. Pass itemId to load the container instead of absolute url
2. Remove other unused interfaces

Remaining changes to the interface - 

We discussed returning the `fileName` to the app developer. The resolvedUrl returns fileName as an empty string. fileName is not part of the current interface, but there will be a follow-up for this change.

[AB#6316](https://dev.azure.com/fluidframework/internal/_workitems/edit/6316)

### CODE REVIEW

File: packages/framework/fluid-static/api-report/fluid-static.api.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ export class DOProviderContainerRuntimeFactory extends BaseContainerRuntimeFacto
 // @public
 export class FluidContainer<TContainerSchema extends ContainerSchema = ContainerSchema> extends TypedEventEmitter<IFluidContainerEvents> implements IFluidContainer<TContainerSchema> {
     constructor(container: IContainer, rootDataObject: IRootDataObject);
-    attach(): Promise<string>;
+    attach(filePath?: string, fileName?: string): Promise<string>;
```
Comment (anthony-murphy) [inline packages/framework/fluid-static/api-report/fluid-static.api.md:43]: i don't think this is the correct way to expose this. i would remove it from this pr, and do a separate PR that only does this. the problem you will have to solve is that this type is also used by the azure client, but it does not take these parametere

Comment (anthony-murphy) [reply packages/framework/fluid-static/api-report/fluid-static.api.md:43]: also, exposing as an object, rather than splayed params will likely be a better path, as it is more supportable and maintainable

Comment (anthony-murphy) [reply packages/framework/fluid-static/api-report/fluid-static.api.md:43]: another change we should make, but no in this PR, is to make the return type of attach an object, not a string, this will allow us to return named things, like {itemId}, {containerId} in a way this is also extendable such that we could expose more data here if we have use, the raw string it too restrictive.

File: packages/service-clients/odsp-client/src/interfaces.ts [hunk 2]
```diff
@@ -30,11 +30,21 @@ export interface OdspConnectionConfig {
 	 * RaaS Drive Id of the tenant where Fluid containers are created
 	 */
 	driveId: string;
+}
 
+/**
+ * @alpha
+ */
+export interface OdspContainerAttachProps {
```
Comment (anthony-murphy) [inline packages/service-clients/odsp-client/src/interfaces.ts:38]: this doesn't appear to be used, can it just be removed for now

File: packages/service-clients/odsp-client/src/odspClient.ts [hunk 3]
```diff
@@ -146,53 +149,42 @@ export class OdspClient {
...
+				uuid(),
+			);
 			if (container.attachState !== AttachState.Detached) {
 				throw new Error("Cannot attach container. Container is not in detached state");
 			}
-			await container.attach(createNewRequest);
-			const absoluteUrl = await container.getAbsoluteUrl("/");
-			if (absoluteUrl === undefined) {
-				throw new Error("Absolute Url not avaiable on attached container");
+			const resolvedUrl = container.resolvedUrl as IOdspResolvedUrl;
```
Comment (anthony-murphy) [inline packages/service-clients/odsp-client/src/odspClient.ts:168]: use getOdspResolvedUrl

Comment (anthony-murphy) [reply packages/service-clients/odsp-client/src/odspClient.ts:168]: we should also never use `as` casting

File: packages/service-clients/odsp-client/src/odspClient.ts [hunk 4]
```diff
@@ -97,14 +91,21 @@ export class OdspClient {
...
 		services: OdspContainerServices;
 	}> {
 		const loader = this.createLoader(containerSchema);
-		const container = await loader.resolve({ url: sharingUrl });
+		const url = createOdspUrl({
+			siteUrl: this.properties.connection.siteUrl,
+			driveId: this.properties.connection.driveId,
+			itemId: id,
+			dataStorePath: "",
+			containerPackageName: "no-dynamic-package",
```
Comment (anthony-murphy) [inline packages/service-clients/odsp-client/src/odspClient.ts:106]: i don't think this need to be specified

Comment (anthony-murphy) [reply packages/service-clients/odsp-client/src/odspClient.ts:106]:
```suggestion
```

File: packages/service-clients/odsp-client/src/odspClient.ts (hunk 3 above)
Comment (sonalideshpandemsft) [reply packages/service-clients/odsp-client/src/odspClient.ts]: can `getOdspResolvedUrl` be marked as public? api extractor complains that there is no release tag added?

Comment (anthony-murphy) [reply packages/service-clients/odsp-client/src/odspClient.ts]: i see. a better change might be to add a new function that is actually a type guard
``` typescript
function isOdspResolveUrl(maybe: IResolvedUrl): maybe is IOdspResolvedUrl{
return "odspResolvedUrl" in maybe;
}
```
export that function, and use it in the existing function, as you probably don't want an assert on failure

File: packages/service-clients/odsp-client/src/odspClient.ts (hunk 4 above)
Comment (sonalideshpandemsft) [reply packages/service-clients/odsp-client/src/odspClient.ts]: removed

File: packages/drivers/odsp-driver/src/odspUtils.ts [hunk 5]
```diff
@@ -305,6 +305,13 @@ export function getOdspResolvedUrl(resolvedUrl: IResolvedUrl): IOdspResolvedUrl
 	return resolvedUrl as IOdspResolvedUrl;
 }
 
+/**
+ * @public
+ */
+export function isOdspResolvedUrl(resolvedUrl: IResolvedUrl): resolvedUrl is IOdspResolvedUrl {
+	return "odspResolvedUrl" in resolvedUrl;
```
Comment (anthony-murphy) [inline packages/drivers/odsp-driver/src/odspUtils.ts:312]:
```suggestion
	return "odspResolvedUrl" in resolvedUrl && resolvedUrl.odspResolvedUrl===true;
```

**Review by anthony-murphy — APPROVED** [review]

File: packages/drivers/odsp-driver/src/odspUtils.ts (hunk 5 above)
Comment (anthony-murphy) [reply packages/drivers/odsp-driver/src/odspUtils.ts:312]: can you also change the implementation of getOdspResolvedUrl to use this new function
